pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -232,6 +276,27 @@ public void acknowledgeMessage(List<Position> positions, AckType ackType, Map<St
             if (log.isDebugEnabled()) {
                 log.debug("[{}][{}] Individual acks on {}", topicName, subName, positions);
             }
+
+            synchronized (PersistentSubscription.this) {
```

Comment:
add this block under an if check statement. so this code will not be processed when transaction is not used.

```
if (pendingAckMessages != null || pendingCumulativeAckMessage != null) {
    // process the logic
}
```